.circleci/config.yml

```diff
@@ -141,6 +209,10 @@ workflows:
     jobs:
       - test_python
       - test_web
+      - test_e2e:
+          requires:
+            - test_python
+            - test_web
       - release:
           requires:
             - test_web
```

Comment:
These `requires` statements need to be changed to `test_e2e` instead of `web` and `python`. As is, the release step might run successfully before the `test_e2e` step can finish, which would be very bad.

Same change for `publish`.

Reply:
Do you mean keep `- test_e2e` in the jobs block, then add `- test_e2e` to both `release` and `publish` 's `requires` block?

Reply:
```   
    jobs:
      - test_python
      - test_web
      - test_e2e:
          requires:
            - test_python
            - test_web
      - release:
          requires:
            - test_e2e
          filters:
            branches:
              only: master
      - publish:
          requires:
            - test_e2e
          filters:
            branches:
              only: master
```
We want a chain of jobs that looks like this:
```
test_web    ---> test_e2e --> release
test_python -/             \- publish
```
It just complicates the graph to make `release` and `publish` require `test_web` and `test_python` when in fact we only need to wait for `test_e2e` to finish.